### PR TITLE
Pass `browser_identity` on to privacy requests from consent request patches

### PR DESCRIPTION
**Pull request by allisonking**
Backend part of https://github.com/ethyca/fidesplus/issues/519

### Code Changes

* [x] Update the Identity model to support a new `user_id` 
* [x] Update the `providedidentitytype` enum to also support `user_id`
* [x] Update the patch endpoint to take a `browser_identity`
* [x] Update the `queue_privacy_request_to_propagate_consent` to merge identities 

### Steps to Confirm

* [ ] _list any manual steps for reviewers to confirm the changes_

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

_Write some things here about the changes and any potential caveats_

### CODE REVIEW

**Comment by allisonking** [comment]
To verify the alembic migration, this query is useful:

```
select n.nspname as enum_schema,  
       t.typname as enum_name,  
       e.enumlabel as enum_value
from pg_type t 
   join pg_enum e on t.oid = e.enumtypid  
   join pg_catalog.pg_namespace n ON n.oid = t.typnamespace WHERE t.typname='providedidentitytype';
```

should yield 

```
 enum_schema |      enum_name       |  enum_value  
-------------+----------------------+--------------
 public      | providedidentitytype | email
 public      | providedidentitytype | phone_number
```
before migration, and after migration should have an additional `user_id` field. The downgrade should work too

File: src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py
Comment (pattisdr) [inline src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py:20]: extra print statement

File: tests/ops/models/test_consent_request.py
Comment (pattisdr) [inline tests/ops/models/test_consent_request.py:230]: This should be ` provided_identity.delete(db)` instead, I realize I made this same mistake above too.

File: src/fides/api/ops/api/v1/endpoints/consent_request_endpoints.py
Comment (pattisdr) [inline src/fides/api/ops/api/v1/endpoints/consent_request_endpoints.py:243]: That's a good way to do this!

File: src/fides/api/ops/schemas/redis_cache.py
Comment (pattisdr) [inline src/fides/api/ops/schemas/redis_cache.py:14]: Have we confirmed this is the user_id that's needed here?  https://support.google.com/analytics/answer/3123669?hl=en, I just assumed that in the design docs, but I don't know for sure

File: tests/ops/models/test_consent_request.py
Comment (pattisdr) [inline tests/ops/models/test_consent_request.py:228]: This is a great test.  Another good test to add would be to the test consent request endpoints `test_set_consent_consent_preferences` and pass in a browser_identity dict and then assert that the user_id was added to identities retrieved in `consent_request.privacy_request.get_persisted_identity()`

File: src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py
Comment (pattisdr) [inline src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py:19]: Are these needed specifically? From my understanding, they automatically commit all the migrations before this operation, potentially prematurely.

> Warning As is necessary, the database transaction preceding the block is unconditionally committed. This means that the run of migrations preceding the operation will be committed, before the overall migration operation is complete.
It is recommended that when an application includes migrations with “autocommit” blocks, that [EnvironmentContext.transaction_per_migration](https://alembic.sqlalchemy.org/en/latest/api/runtime.html#alembic.runtime.environment.EnvironmentContext.params.transaction_per_migration) be used so that the calling environment is tuned to expect short per-file migrations whether or not one of them has an autocommit block.

**Review by pattisdr — COMMENTED** [review]
Bravo, thanks for knocking this out so quickly - just a few small changes requested

File: CHANGELOG.md
Comment (pattisdr) [inline CHANGELOG.md:41]: All these changelog items will need to be removed to unreleased at some point - but this can also wait until we merge `consent-propagation` next week too

File: src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py
Comment (allisonking) [reply src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py:20]: 🤦‍♀️ at least it wasn't more embarrassing

Comment (pattisdr) [reply src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py:20]: 😄 I know what you mean

**Comment by pattisdr** [comment]
hey @allisonking just a heads-up that I updated the base `consent-propagation` branch with main after `unified-fides-resources` was merged because there were several conflicts there.  

I think the main consequences for you when getting up to date with `consent-propagation` are just the changelog and bumping the downrev of your migration.

File: src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py
Comment (allisonking) [reply src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py:19]: apparently the `ALTER TYPE` command [cannot be executed in a transaction](https://www.postgresql.org/docs/9.1/sql-altertype.html#AEN63173). I don't 100% understand it, but these answers go into some more detail: https://stackoverflow.com/questions/14845203/altering-an-enum-field-using-alembic

Comment (pattisdr) [reply src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py:19]: That seems like it was for older postgres versions though

https://www.postgresql.org/docs/12/sql-altertype.html
> If ALTER TYPE ... ADD VALUE (the form that adds a new value to an enum type) is executed inside a transaction block, the new value cannot be used until after the transaction has been committed.

We do this in other migrations:
https://github.com/ethyca/fides/blob/main/src/fides/api/ctl/migrations/versions/906d7198df28_privacy_request_approve.py#L20

Comment (allisonking) [reply src/fides/api/ctl/migrations/versions/392992c7733a_add_user_id_as_a_provided_identity_type.py:19]: oh great, thanks for finding that!!

**Comment by allisonking** [comment]
Thanks, Dawn! I think I've addressed your comments now.

> I think the main consequences for you when getting up to date with consent-propagation are just the changelog and bumping the downrev of your migration.

I didn't see any alembic conflicts (didn't find another alembic file using that same downrev, and `alembic heads` only returned the one head) but let me know if I'm missing something!

**Review by pattisdr — APPROVED** [review]
> I didn't see any alembic conflicts (didn't find another alembic file using that same downrev, and alembic heads only returned the one head) but let me know if I'm missing something!

Whoops you're right, your migration is already pointed at `de456534dbda`, sorry!

This looks ready to merge into `consent-propagation`: thanks especially for the extra assertion in the consent request endpoints test.

**Pull request merged**
